### PR TITLE
Add mgos_i2c_{get,set}bits_reg_{w,b}()

### PR DESCRIPTION
**Pull request by pimvanpelt**
See comment for the *proof*. It's quite unorthodox to add unit tests to `mongoose-os-libs`, but I do have unit tests in my home client and happy to share them.
They test the boundary conditions:
*   bitlen=0 -> FAIL
*   bitlen=1 && bitoffset=16 -> PASS
*   bitlen=2 && bitoffset=16 -> FAIL
*   bitlen=3 && value=7 -> PASS
*   bitlen=3 && value=8 -> FAIL
*   bitoffset=17 -> FAIL

And they also test normal writing conditions thoroughly.

@mamuesp FYI
@rojer please review / merge if you like the semantics here.

### CODE REVIEW

**Comment by pimvanpelt** [comment]
Also field tested this on `mcp23xxx` library, which fiddles with bitmasks to enable and acknowledge interrupts on the GPIO input pins. Works on ESP8266 and ESP32 at least.

File: include/mgos_i2c.h [hunk 1]
```diff
@@ -142,6 +142,33 @@ bool mgos_i2c_write_reg_w(struct mgos_i2c *conn, uint16_t addr, uint8_t reg,
...
+ * - bitoffset: 0..7 is the position at which to write `value`
+ * - bitlen   : number of bits to write
+ * - value    : the value to write there
+ *
+ * Invariants:
+ * - value must fit in `bitlen` (ie value < 2^bitlen)
+ * - bitlen+bitoffset <= register size (8 for reg_b, 16 for reg_w)
+ * - bitlen cannot be 0.
+ * - *conn cannot be NULL.
+ *
```
Comment (rojer) [inline include/mgos_i2c.h]: can you add a couple examples here and illustrate which bits get twiddled?

Comment (pimvanpelt) [reply include/mgos_i2c.h]: Take a look at https://github.com/mongoose-os-libs/i2c/pull/6/commits/d0d56fd0ce43f246e2597dc4775db30fae608cb6 below.

**Comment by rojer** [comment]
great, thanks, integrating now
